### PR TITLE
Charlesmchen/sae filenames

### PR DESCRIPTION
**Pull request by charlesmchen**
The "coerce to NSData" hack won't work; it prevents us from preserving the file names of shared files.  It's probably a tad safer, but I don't know of a specific share type that fixes at the moment.  I think it's far better to preserve filenames and fix broken share types (if any) when we find them.  They should be rare, if they even exist.

PTAL @michaelkirk

### CODE REVIEW

File: SignalShareExtension/ShareViewController.swift [hunk 1]
```diff
@@ -713,7 +710,7 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
                 } else {
                     return itemUrl
                 }
-            }()
+                }()
```
Comment (michaelkirk) [inline SignalShareExtension/ShareViewController.swift]: weird indent

File: SignalShareExtension/ShareViewController.swift [hunk 2]
```diff
@@ -91,7 +91,7 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
 
             Logger.debug("\(strongSelf.logTag) setup is slow - showing loading screen")
             strongSelf.showPrimaryViewController(loadViewController)
-        }.retainUntilComplete()
+            }.retainUntilComplete()
```
Comment (michaelkirk) [inline SignalShareExtension/ShareViewController.swift]: weird indent

File: SignalShareExtension/ShareViewController.swift [hunk 3]
```diff
@@ -494,19 +494,19 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
             strongSelf.loadViewController = nil
             strongSelf.showPrimaryViewController(conversationPicker)
             Logger.info("showing picker with attachment: \(attachment)")
-        }.catch {[weak self]  error in
-            AssertIsOnMainThread()
-            guard let strongSelf = self else { return }
+            }.catch {[weak self]  error in
```
Comment (michaelkirk) [inline SignalShareExtension/ShareViewController.swift]: weird indent

**Comment by charlesmchen** [comment]
Does this also fix the ability to share urls from safari/chrome?

@michaelkirk Yes

File: SignalShareExtension/ShareViewController.swift (hunk 1 above)
Comment (charlesmchen) [reply SignalShareExtension/ShareViewController.swift]: Fixed.

File: SignalShareExtension/ShareViewController.swift (hunk 2 above)
Comment (charlesmchen) [reply SignalShareExtension/ShareViewController.swift]: Fixed.

File: SignalShareExtension/ShareViewController.swift (hunk 3 above)
Comment (charlesmchen) [reply SignalShareExtension/ShareViewController.swift]: Done.

**Pull request merged**